files/base.rb

```diff
@@ -222,6 +223,10 @@ def handler_settings
     settings[settings_key]
   end
 
+  def reset_api_settings!
+    @settings['api'].merge!(settings['api_client']) if settings['api_client']
```

Comment:
Technically yes even though right now it will work if we drop "@".

https://github.com/sensu-plugins/sensu-plugin/blob/master/lib/sensu-plugin/utils.rb#L19

After `if` we have to use `settings` to ensure `@settings` is populated first.

I call `merge!` on `@settings` instead of `settings` because in upstream it's a read method "in spirit" while `@settings` is the actual mutable thing that holds the entire hash.